tools/nl/svindex_differ/differ.py

```diff
@@ -67,8 +114,48 @@ def _maybe_copy_embeddings(file):
   return file
 
 
-def _diff_table(base, test):
-  return difflib.HtmlDiff().make_table(base, test)
+def _get_diff_table(diff_list, base_sv_info, test_sv_info):
+  """Given a list of diffs produced by the difflib.Differ, get the rows of
+  sv information to show in the diff table.
+  """
+  diff_table_rows = []
+  last_added = -1
+  for i, diff in enumerate(diff_list):
+    if i <= last_added:
+      continue
+    last_added = i
+    # difflib.Differ will add 2 characters in front of the original text for every line.
+    diff_sv = diff[2:]
+    next_diff = None
+    if i < len(diff_list) - 1:
+      next_diff = diff_list[i + 1]
+    # If theres no + or -, that means this line is the same in both base and test.
+    if not diff.startswith('+') and not diff.startswith('-'):
+      info = base_sv_info.get(diff_sv, test_sv_info.get(diff_sv))
+      diff_table_rows.append((info, info))
+    # If the line starts with -, this means it was present in base but not in test.
+    elif diff.startswith('-'):
+      base_info = base_sv_info.get(diff_sv, {})
+      base_info['color'] = _SUB_COLOR
+      test_info = {}
+      if next_diff and next_diff.startswith('+'):
+        test_sv = next_diff[2:]
+        test_info = test_sv_info.get(test_sv, {})
+        test_info['color'] = _ADD_COLOR
+        last_added = i + 1
+      diff_table_rows.append((base_info, test_info))
+    # Otherwise, the line started with +, which means it was present in test but not base.
+    else:
+      base_info = {}
+      test_info = test_sv_info.get(diff_sv, {})
+      test_info['color'] = _ADD_COLOR
+      if next_diff and next_diff.startswith('-'):
+        base_sv = next_diff[2:]
+        base_info = base_sv_info.get(base_sv, {})
+        base_info['color'] = _SUB_COLOR
+        last_added = i + 1
+      diff_table_rows.append((base_info, test_info))
+  return diff_table_rows
```

Comment:
Wonder if its better to make the table show SV name instead of DCID by default?

Reply:
I tried it with the sv name and found it easier to scan the dcids, but I updated it so that the name is outside the details section, which looks like

<img width="1276" alt="Screenshot 2023-11-08 at 8 05 00 AM" src="https://github.com/datacommonsorg/website/assets/69875368/c6684d38-54fb-495c-b066-64da5ee220e2">